basiclibrary/src/main/java/inheritance/Review.java

```diff
@@ -2,39 +2,69 @@
 
 public class Review {
     String user;
-    String userReview;
-    String restaurantName;
+    String body;
+    String movie;
+    Restaurant restaurant;
+    Shop shop;
+    Theater theater;
     int stars;
 
-    public Review(String user, String userReview, int stars){
+    //constructor for Restaurant reviews
+    public Review(String user, String body, int stars, Restaurant restaurant){
```

Comment:
It might have been a good idea in each of these constructors to set the other two classes that aren't being assigned a value to null. So in this constructor:
`this.shop = null`
`this.theater = null`